extensions/pkg/predicate/predicate_test.go

```diff
@@ -87,6 +89,72 @@ var _ = Describe("Predicate", func() {
 		})
 	})
 
+	Describe("#AddTypePredicate", func() {
+		var (
+			object       client.Object
+			createEvent  event.CreateEvent
+			updateEvent  event.UpdateEvent
+			deleteEvent  event.DeleteEvent
+			genericEvent event.GenericEvent
+		)
+
+		BeforeEach(func() {
+			object = &extensionsv1alpha1.Extension{
+				Spec: extensionsv1alpha1.ExtensionSpec{
+					DefaultSpec: extensionsv1alpha1.DefaultSpec{
+						Type: extensionType,
+					},
+				},
+			}
+			createEvent = event.CreateEvent{
+				Object: object,
+			}
+			updateEvent = event.UpdateEvent{
+				ObjectOld: object,
+				ObjectNew: object,
+			}
+			deleteEvent = event.DeleteEvent{
+				Object: object,
+			}
+			genericEvent = event.GenericEvent{
+				Object: object,
+			}
+		})
+
+		It("should return true because only one extention exists and extention type matches", func() {
+			predicates := AddTypePredicate([]predicate.Predicate{}, extensionType)
+
+			Expect(len(predicates)).To(Equal(1))
+
+			Expect(predicates[0].Create(createEvent)).To(BeTrue())
+			Expect(predicates[0].Update(updateEvent)).To(BeTrue())
+			Expect(predicates[0].Delete(deleteEvent)).To(BeTrue())
+			Expect(predicates[0].Generic(genericEvent)).To(BeTrue())
+		})
+
+		It("or predicate should return true because one of extention type match", func() {
+			predicates := AddTypePredicate([]predicate.Predicate{}, extensionType, "other")
+
+			Expect(len(predicates)).To(Equal(1))
+
+			Expect(predicates[0].Create(createEvent)).To(BeTrue())
+			Expect(predicates[0].Update(updateEvent)).To(BeTrue())
+			Expect(predicates[0].Delete(deleteEvent)).To(BeTrue())
+			Expect(predicates[0].Generic(genericEvent)).To(BeTrue())
+		})
+
+		It("or predicate should return false because none of the extention type matches", func() {
+			predicates := AddTypePredicate([]predicate.Predicate{}, "foo", "bar")
+
+			Expect(len(predicates)).To(Equal(1))
+
+			Expect(predicates[0].Create(createEvent)).To(BeFalse())
+			Expect(predicates[0].Update(updateEvent)).To(BeFalse())
+			Expect(predicates[0].Delete(deleteEvent)).To(BeFalse())
+			Expect(predicates[0].Generic(genericEvent)).To(BeFalse())
+		})
+	})
```

Comment:
Please add a test case where you pass more than one extension type, and check the returned slice has all the `HasType` predicates.

Reply:
Done